src/sagemaker/workflow/emr_step.py

```diff
@@ -198,9 +208,16 @@ def __init__(
         if cluster_id is not None and cluster_config is not None:
             raise ValueError(ERR_STR_WITH_BOTH_CLUSTER_ID_AND_CLUSTER_CFG.format(step_name=name))
 
+        if execution_role_arn is not None and cluster_config is not None:
```

Comment:
Seems `execution_role_arn` has to come along with `cluster_id` and the former is used to allow accessing the `cluster_id` right?

If it's true, then can we make the logic here more intuitive? e.g. check that execution_role_arn can be not None only if cluster_id is not None. 

And it'd be good to improve the error message to make it more informative, e.g. "execution_role_arn should not be set if ... since it's used to ...", so that it can educate the users why the proper usage looks like this.

Reply:
I think the two ways are equally intuitive here. There are two ways for the customer to make an EMR step: by specifying ID of existing cluster to add steps to, or by specifying config of new cluster to be created with steps. EMR runtime roles are only supported on the former request (where [`AddJobFlowSteps` API](https://docs.aws.amazon.com/emr/latest/APIReference/API_AddJobFlowSteps.html) is used) and not on the latter (where [`RunJobFlow` API](https://docs.aws.amazon.com/emr/latest/APIReference/API_RunJobFlow.html) is used).

How much info should the error message include? Should it contain all the above background?

Reply:
I feel it'd be better to include
- The context - what EMR runtime role is used for, i.e. to add step with the existing cluster. 
   - at least we should include this in the sdk doc string as we don't expect customers to carefully explore all related docs, e.g. API docs
- What's wrong - i.e. EMR runtime role should not be set if `cluster_config` is not None. 

Given this context, that's why I feel it'd be better to group `execution_role_arn` validation with `cluster_id`. But as long as we put enough info in the doc string or error message to educate users and future developers, either way is fine.